src/devices/machine/gew12_uart.h

```diff
@@ -0,0 +1,69 @@
+// license:BSD-3-Clause
+// copyright-holders:Devin Acker
+/*********************************************************************
+
+	gew12_uart.h
+
+	Yamaha GEW12 UART
+
+*********************************************************************/
+
+#ifndef MAME_MACHINE_GEW12_UART_H
+#define MAME_MACHINE_GEW12_UART_H
+
+#include "diserial.h"
+
+class gew12_uart_device : public device_t, public device_serial_interface
+{
+public:
```

Comment:
Integrated 6502 peripherals should use the facilities the core provides for this – look at how e.g the MOS 6500/1 implements its counter/latch.  The events should all come off the execute loop so they happen in the same scheduler domain.

Reply:
How doable is this without having (or needing to create) a 65c02 equivalent of `m6502_mcu_device`?

Maybe it'd be easier for now to just get rid of the UART - I mostly implemented it as-is just to document how it works, but it's pretty much useless at the moment since receiving MIDI will make any of those machines just go into an infinite loop trying to use the nonexistent PCM registers.

Reply:
You’d need to create an `m65c02_mcu_device` but that should be easy – just template the base class of `m6502_mcu_device` (move it to its own .h and .ipp) and add a bunch of `this->` so it compiles.

Reply:
Makes sense. I may just remove the UART for the time being though and redo it that way later, if/when I end up turning that driver into something more than a skeleton. The main focus of this PR is just the (actually working) GEW7 stuff anyway.